.github/workflows/uploadFirrtlReleaseArtifacts.yml

```diff
@@ -36,45 +36,52 @@ jobs:
           files: circt-full-sources.tar.gz*
           repo-token: ${{ secrets.GITHUB_TOKEN }}
 
+  # This job determines what the build matrix should be.  If this is a
+  # "scheduled" or "workflow dispatch" job, then only build Linux images.  If it
+  # is anything else (i.e., release) then build everything.
```

Comment:
AFAICT, these jobs are all run unconditionally while this comment suggests they are only run sometimes. I know follow on work will make it true but it is a little weird.